### ISSUE TITLE
[feature]: linking git commits to issues

### ISSUE THREAD

**Issue by unknowntrojan** [makeplane/plane#2508]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

In JIRA, whenever a commit containing the issue ID in it's message is pushed (example: "PROJ-1 add service X"), it will appear in the JIRA issue's page, showing a clickable link containing the message and author of the commit that would take you to the commit's page inside the git provider (GitHub, GitLab or Gitea).

Plane would need to either periodically read the git repository or receive a webhook from the git software. It would need to know the URL of the git provider for the clickable commit link and needs to know which git user corresponds to which plane user. This would also allow easier per-user tracking inside Plane.

For the 0.14 milestone, OIDC support is planned. This feature could be integrated so that users inside the git are mapped to users from the identity provider.

### Why should this be worked on?

This feature brings plane closer to JIRA's feature set. It also allows more insight into the actual development being done regarding any specific issue.

**Comment by rhea0110** [comment]
Hey @unknowntrojan, thank you for the feature request. We will work on it.

**Comment by LazyCoder-1506** [comment]
Linear does this by linking the GItHub PR to the issue. This is achieved by using the same branch name that is suggested by Linear, and using a webhook to identify the GIthub PR through it's branch name.
But currently this causes one UX problem: the Linear status (Todo/In Progress/Done etc) is a step behind the actual status of the issue, and eventually requires manually updating the status. This is because Linear can't recognise that a user has created the branch on their local machine, Linear can only rely on the GitHub platform, thus there's a step missing in the workflow

**Comment by vihar** [comment]
Thank you for the detailed explanation, @LazyCoder-1506. I agree that this issue should be addressed within Plane. As @vamsi is currently working on integrations, I am tagging him to further discuss and explore potential solutions.

**Comment by rudacs** [comment]
I would like to reinforce how important this feature is for better project organization, control, and traceability.

In Jira, we can easily see whether commits, pull requests, or merge requests are already linked to an issue. This gives the team a much clearer view of the actual development progress without needing to manually search across Git providers.

For teams using Plane as an alternative to Jira, this is not just a nice-to-have integration. It is essential for day-to-day workflow and very useful for controlling the development process, especially when tracking whether work has actually started, whether code was pushed, and whether a change is already under review or merged.

This request has been open since 2023, and I believe Plane should prioritize investing in this. Native support for linking commits and pull/merge requests to issues would make Plane much stronger as a serious project management tool for software teams.

**Comment by manubuss** [comment]
Yes, I also would need this feature and its standard in other tools like jira and linear.

**Comment by vihar** [comment]
PR linking shipped. Reference a work item in a GitHub PR title or description — `[WEB-344]` gives you full state automation, `WEB-344` without brackets just creates a reference link — and Plane moves the work item through your mapped states as the PR goes draft → open → review → merged. It also posts a confirmation comment back on the PR: https://docs.plane.so/integrations/github — on Pro.

Commit-message linking specifically isn't supported; it's PR title and description only.

Closing this out.
